### PR TITLE
[Typing][C-120] Add type annotations for `python/paddle/quantization/base_quanter.py`

### PR DESCRIPTION
**Pull request by enkilee**
### PR Category

User Experience

### PR Types

Improvements

### Description

- https://github.com/PaddlePaddle/Paddle/issues/65008
C120

### CODE REVIEW

**Review by megemini — APPROVED** [review]
LGTM ~ 🤟

**Review by luotao1 — DISMISSED** [review]

File: python/paddle/quantization/base_quanter.py [hunk 1]
```diff
@@ -11,48 +11,50 @@
...
         It can be none which means the quanter didn't hold zero points for quantization.
         """
         pass
 
     @abc.abstractmethod
-    def quant_axis(self) -> Union[int, Iterable]:
+    def quant_axis(self) -> int | Iterable[int]:
         r"""
         Get the axis of quantization. None means tensor-wise quantization.
         """
```
Comment (SigureMo) [inline python/paddle/quantization/base_quanter.py:60]: 下面的 `bit_length` 也加一下？

Comment (enkilee) [reply python/paddle/quantization/base_quanter.py:60]: 收到

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**